mlflow/tracking/context/registry.py

```diff
@@ -68,9 +72,11 @@ def resolve_tags(tags=None):
 
     all_tags = {}
     for provider in _run_context_provider_registry:
-        if provider.in_context():
-            # TODO: Error out gracefully if provider's tags are not valid or have wrong types.
-            all_tags.update(provider.tags())
+        try:
+            if provider.in_context():
+                all_tags.update(provider.tags())
+        except Exception as e: # pylint: disable=broad-except
```

Comment:
@dbczumar I left this catch a broad exception. My rationale is we don't know what kind of failure might arise from different context provider's implementation. Therefore disabled `pylint: disable=broad-except`

Reply:
Sounds great!